### PR TITLE
Player and monster class functions

### PR DESCRIPTION
**Pull request by filiptrplan**
Added basic functions to Monster and Player classes just so they can be called by different features. I haven't written any logic in the functions.
Added relative setters such as `addScore` for the relevant variables so we can call functions on every change of the variable. The setters are relative because it is easier to understand. 

Also changed all the String and Number interface type variables to their primitive counterparts to avoid any confusion.
Fixed all the compile bugs of the `world` folder

### CODE REVIEW

File: world-core/src/router/Router.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class Router {
         }
 
         catch (e) {
-            Logger.warn(`Problem emitting message '${msg.type}'`, { exception: e.toString() });
+            Logger.warn(`Problem emitting message '${msg.type}'`, { exception: e.tostring() });
```
Comment (davidmohar) [inline world-core/src/router/Router.ts]: This is odd, it should be `.toString()` on exceptions. Was this not working for you? What was the error message? Same question for `world-server/src/network/Server.ts` change as well.

**Review by davidmohar — CHANGES_REQUESTED** [review]

File: world-core/src/router/Router.ts (hunk 1 above)
Comment (filiptrplan) [reply world-core/src/router/Router.ts]: Just an artifact of find and replace. Fixed it

**Review by davidmohar — DISMISSED** [review]

**Comment by davidmohar** [comment]
@1234filip there are merge conflicts, please rebase and resolve

**Comment by filiptrplan** [comment]
Merged develop and solved the conflicts

**Review by davidmohar — APPROVED** [review]

**Pull request merged**